docs/advanced_topics/documents/custom_document_model.md

```diff
@@ -0,0 +1,68 @@
+(custom_document_model)=
+
+# [Custom document model](custom_document_model)
```

Comment:
This here is not needed, just use a normal title
```suggestion
# Custom document model
```

As per other comments, this will convert the header to a link to itself which may break the parser.

There are a few things going on  here.
1. `(custom_document_model)=` - this tells MyST to create a ref to this next 'header' called `custom_document_model`
2. RST will also create a link to headers
3. There will be an implicit link to all documents via their page name, so possibly this ref is not even needed, other places could just link to the page not the ref - but that is not a big deal, we are using this approach in  lots of places.